sql/parser/sql.y

```diff
@@ -1190,21 +1199,42 @@ table_constraint:
 
 constraint_elem:
   CHECK '(' a_expr ')' {}
-| UNIQUE '(' name_list ')'
+| UNIQUE '(' name_list ')' opt_storing
   {
-    $$ = &IndexTableDef{Unique: true, Columns: NameList($3)}
+    $$ = &IndexTableDef{Unique: true, Columns: NameList($3), Storing: $5}
   }
-| UNIQUE existing_index {}
 | PRIMARY KEY '(' name_list ')'
   {
     $$ = &IndexTableDef{PrimaryKey: true, Unique: true, Columns: NameList($4)}
   }
-| PRIMARY KEY existing_index {}
 | EXCLUDE '(' exclusion_constraint_list ')'
     exclusion_where_clause {}
 | FOREIGN KEY '(' name_list ')' REFERENCES qualified_name
     opt_column_list key_match key_actions {}
 
+storing:
+  COVERING
+| STORING
+
+// TODO(pmattis): It would be nice to support a syntax like STORING
+// ALL or STORING (*). The syntax addition is straightforward, but we
+// need to be careful with the rest of the implementation. In
+// particular, columns stored at indexes are currently encoded in such
+// a way that adding a new column would require rewriting the existing
+// index values. We will need to change the storage format so that it
+// is a list of <columnID, value> pairs which will allow both adding
+// and dropping columns without rewriting indexes that are storing the
```

Comment:
Of course, this strategy runs into difficulties if the default value is changed.